docs/source/framework.rst

```diff
@@ -174,6 +174,39 @@ This provide a base class that will behave as normal :code:`DataContainer`.
    :members:
    :inherited-members:
 
+Partitioner
+===========
+
+This class allows a user to partition an instance of :code:`AcquisitionData` into a number of subsets. For example, to use with a stochastic optimisation method. 
+
+For example: 
+
+.. code-block :: python
+
+   from cil.utilities import dataexample
+   from cil.plugins.astra.operators import ProjectionOperator
+   
+   # get the data  
+   data = dataexample.SIMULATED_PARALLEL_BEAM_DATA.get()
+   data.reorder('astra')
+   data = data.get_slice(vertical='centre')
+
+   # create the geometries 
+   ag = data.geometry 
+   ig = ag.get_ImageGeometry()
+
+   # partition the data and build the projectors
+   n_batches = 10 
+   partitioned_data = data.partition(num_batches=n_batches, mode='sequential') # Choose mode from `sequential`, `staggered` or `random_permutation` 
```

Comment:
I think unnecessary to define extra variable n_batches=10, clearer to pass 10 directly. Or if preferred define also the mode string as a named variable before this line, to pass both n_batches and the mode as inputs for consistency.

Reply:
Done